crowdin.js

```diff
@@ -0,0 +1,178 @@
+const AdmZip = require('adm-zip');
+const program = require('commander');
+const request = require('request');
+const fs = require('fs');
+
+/* eslint-disable no-console */
+
+function help() {
+	console.log('To use crowdin script you need API key. You can set this key under .crowdin.json file { "key": "..." } or use -k');
+	console.log('To use crowdin script to get the status: ');
+	console.log('>node crowding.js --status');
+	console.log('To use crowdin script to download current locales: ');
+	console.log('>node crowding.js --download');
+}
+
+program
+	.version('0.0.1')
+	.option('-k, --key [value]', 'Required API key (or use .crowdin.json)')
+	.option('-v, --verbose', 'display debug info')
+	.option('-p, --project [value]', 'PROJECT_ID to set API URL. Default talendui')
+	.option('-d, --download', 'download all the translations')
+	.option('-s, --status', 'get the status of translations')
+	.option('-u, --upload', 'upload translation')
+	;
+
+program.on('--help', help);
+
+program.parse(process.argv);
+
+function debug(...args) {
+	if (program.verbose) {
+		console.log(...args);
+	}
+}
+
+// TODO: prompt for the API_KEY
+
+let API_KEY = program.key;
+let CROWDIN;
+if (fs.existsSync('.crowdin.json')) {
+	// eslint-disable-next-line
+	CROWDIN = require('./.crowdin.json');
+	if (CROWDIN.key) {
+		API_KEY = CROWDIN.key;
+	}
+}
+
+const PROJECT_ID = program.project || CROWDIN.project || 'talendui';
+const URL = `https://api.crowdin.com/api/project/${PROJECT_ID}`;
+
+if (!API_KEY) {
+	throw new Error('you must set -k or put a key in .crowdin config file');
+}
+
+const FILES = [
+	{
+		name: 'tui-components.json',
+		path: `${__dirname}/i18n/components/en/tui-components.json`,
```

Comment:
we can store the template on each packages in packages/{{package}}/locales/template/{{namespace}}.

So, we can add some variants for a key which are not parsed by i18next-parser (plural, context, ...)